test/python/transpiler/test_dynamical_decoupling.py

```diff
@@ -622,6 +622,90 @@ def test_dd_with_calibrations_with_parameters(self, param_value):
+        q_0: ──────┤ H ├─────────■──┤ Delay(40[dt]) ├──────┤ X ├──────┤ Delay(70[dt]) ├»
+             ┌─────┴───┴─────┐ ┌─┴─┐└───────────────┘┌─────┴───┴─────┐└─────┬───┬─────┘»
+        q_1: ┤ Delay(50[dt]) ├─┤ X ├────────■────────┤ Delay(20[dt]) ├──────┤ X ├──────»
+             ├───────────────┴┐└───┘      ┌─┴─┐      └───────────────┘      └───┘      »
+        q_2: ┤ Delay(750[dt]) ├───────────┤ X ├──────────────■─────────────────────────»
+             ├────────────────┤           └───┘            ┌─┴─┐                       »
+        q_3: ┤ Delay(950[dt]) ├────────────────────────────┤ X ├───────────────────────»
+             └────────────────┘                            └───┘                       »
+        «           ┌───┐      ┌───────────────┐      ┌───┐      ┌───────────────┐»
+        «q_0: ──────┤ Y ├──────┤ Delay(70[dt]) ├──────┤ X ├──────┤ Delay(70[dt]) ├»
+        «     ┌─────┴───┴─────┐└─────┬───┬─────┘┌─────┴───┴─────┐└─────┬───┬─────┘»
+        «q_1: ┤ Delay(20[dt]) ├──────┤ Y ├──────┤ Delay(20[dt]) ├──────┤ X ├──────»
+        «     └───────────────┘      └───┘      └───────────────┘      └───┘      »
+        «q_2: ────────────────────────────────────────────────────────────────────»
+        «                                                                         »
+        «q_3: ────────────────────────────────────────────────────────────────────»
+        «                                                                         »
+        «           ┌───┐      ┌───────────────┐
+        «q_0: ──────┤ Y ├──────┤ Delay(50[dt]) ├─────────────────
+        «     ┌─────┴───┴─────┐└─────┬───┬─────┘┌───────────────┐
+        «q_1: ┤ Delay(20[dt]) ├──────┤ Y ├──────┤ Delay(20[dt]) ├
+        «     └───────────────┘      └───┘      └───────────────┘
+        «q_2: ───────────────────────────────────────────────────
+        «
+        «q_3: ───────────────────────────────────────────────────
+        «
+        """
+        dd_sequence = [XGate(), YGate(), XGate(), YGate()]
+        pm = PassManager(
+            [
+                ALAPSchedule(self.durations),
+                DynamicalDecoupling(self.durations, dd_sequence, pulse_alignment=10),
+            ]
+        )
+
+        ghz4_dd = pm.run(self.ghz4)
+
+        expected = self.ghz4.copy()
+        expected = expected.compose(Delay(50), [1], front=True)
+        expected = expected.compose(Delay(750), [2], front=True)
+        expected = expected.compose(Delay(950), [3], front=True)
+
+        expected = expected.compose(Delay(40), [0])
+        expected = expected.compose(XGate(), [0])
+        expected = expected.compose(Delay(70), [0])
+        expected = expected.compose(YGate(), [0])
+        expected = expected.compose(Delay(70), [0])
+        expected = expected.compose(XGate(), [0])
+        expected = expected.compose(Delay(70), [0])
+        expected = expected.compose(YGate(), [0])
+        expected = expected.compose(Delay(50), [0])
```

Comment:
so here len(slack)=300. Why is it not distributed as 5 `Delay(60)` operations?

Reply:
This is how default spacing is designed, i.e. edge delays are half of ones in the middle.

In the original design: edge = 37.5 dt, mid = 75 dt

The nearest pattern satisfying the constraints is 40, 70, ..., 70, 50 rather than 60, 60, ..., 60, 60, but this depends how you want to design the default sequence.

Reply:
i see yes i remember now the default was [d/2, d, .., d, d/2]